packages/seacas/libraries/ioss/src/Ioss_DatabaseIO.h

```diff
@@ -246,6 +246,16 @@ namespace Ioss {
       flush_database_nl();
     }
 
+    /** \brief Checks if a database type supports groups
+     *
+     *  \returns True if successful.
+     */
+    bool supports_group()
```

Comment:
I think this is exposing the method that we are using in exodus through netCDF -- `groups` as a general concept in the IOSS API.  

I think that instead the client knows that their output/model is going to have topology changes and want to store that in the best way possible.  One way is that they know they are using exodus as the output format and they either want single file output or each change in a separate file.  They don't care about how this is done internally in the file.

I don't know what our API-visible interface for this should be, but `groups` is very overloaded term and could be confused with `assemblies`.  Or, to put another way, `groups` doesn't suggest to me any association with changing topology.  

It may be ok to use `group` down at the exodus level, but if cgns supports something like `delta-sets` as their method for topology changes then we have confusion in the API...  

I'm not sure what to suggest, but seeing all this group-related api functions at this level just seems wrong...

Maybe something like `change-set` or ?   -- All database type could/might support this.  In some it is a set of appropriately-named files; in some it uses the netCDF groups, if some other type each file contains full model topology and there is some metadata file that says what happened from file to file...

At creation, the client specifies that topology changes will happen, so IOSS does the appropriate action (whatever that might be).  On read, IOSS queries the database(s) and can report that there are topology changess or multiple "change-sets".

Note that the "change-sets" might all contain the same topology -- is this is a run with multiple restarts, each restart state might be in a separate "change-set" even though the topology from one to another is the same.

Reply:
I think we have the basics of this working in exodus at this point, but now need to plan for how the client interfaces with the concept and how the concept gets expanded out as a "database-agnostic" concept that encompasses and encapsulates as much similar behavior as possible. It also needs to plan for extension.

I don't know what that API view looks like, but I don't think "groups" is the correct   one.

Reply:
@gdsjaar I think we probably need to meet to discuss  these concepts and nail down what the API should eventually look like